src/pages/product.js

```diff
@@ -16,13 +16,14 @@ class ProductDetailPage extends Component {
     /**
      * Catalog Product item
      */
-    product: PropTypes.object.isRequired
+    product: PropTypes.object.isRequired,
+    shop: PropTypes.object.isRequired
   }
 
   render() {
     return (
       <Layout>
-        <ProductDetail product={this.props.product} />
+        <ProductDetail product={this.props.product} shop={this.props.shop} />
```

Comment:
I see In some places you have `currencyCode` as a prop, but for the `ProductDetail` component you're passing in the `shop`. Maybe it would be better to just pass the `currencyCode`?

And if you do pass the currency code alone, then consider making it explicitly required.

```
shop: PropTypes.shape({
  currency: PropTypes.shape({
    code: PropTypes.string.isRequired
  })
})
```